eng/pipelines/stages/build-abstractions-package-ci-stage.yml

```diff
@@ -0,0 +1,351 @@
+################################################################################
+# Licensed to the .NET Foundation under one or more agreements.  The .NET
+# Foundation licenses this file to you under the MIT license.  See the LICENSE
+# file in the project root for more information.
+################################################################################
+
+# This stage builds the Abstractions package, runs tests, and publishes the
+# resulting NuGet packages as pipeline artifacts.
+#
+# The NuGet packages have the following properties:
+#
+#   Name:     Microsoft.Data.SqlClient.Extensions.Abstractions Version:
+#   <packageVersion>.<buildNumber>
+#
+#   Where <packageVersion> and <buildNumber> are the values of the
+#   'packageVersion' and 'buildNumber' parameters, respectively.
+#
+# The following NuGet packages are published:
+#
+#   Microsoft.Data.SqlClient.Extensions.Abstractions.<version>.nupkg
+#   Microsoft.Data.SqlClient.Extensions.Abstractions.<version>.snupkg (symbols)
+#
+# The packages are published to pipeline artifacts with the name specified by
+# the 'artifactName' parameter.
+#
+# This template defines a stage named 'build_abstractions_package_stage' that
+# can be depended on by downstream stages.
+
+parameters:
+  # The type of build to produce (Release or Debug)
+  - name: buildConfiguration
+    displayName: Build Configuration
+    type: string
+    default: Release
+    values:
+    - Release
+    - Debug
+
+  # The version (Major.Minor.Patch) to apply to the package.
+  - name: packageVersion
+    displayName: Package Version Override
+    type: string
+    default: ''
+
+  # The build number of the pipeline.
+  - name: buildNumber
+    displayName: Build Number
+    type: string
+    default: $(Build.BuildNumber)
+
+  # The name of the pipeline artifact to publish.
+  - name: artifactName
+    displayName: Pipeline Artifact Name
+    type: string
+    default: Abstractions.Artifact
+
+  # The list of .NET runtimes to test against.
+  - name: netTestRuntimes
+    displayName: .NET Test Runtimes
+    type: object
+    default: [net8.0, net9.0]
+
+  # The list of .NET Framework runtimes to test against.
+  - name: netFrameworkTestRuntimes
```

Comment:
Not expecting to solve this in this PR, but we should have this list centralized somewhere rather than here, in a directory.build.props, in some csproj files, etc...